File: ETL-Stream-Java/src/main/java/com/aaa/etl/util/US_STATES.java
package com.aaa.etl.util;

public enum US_STATES {

	AK("AK", "Alaska"),
	AL("AL", "Alabama"),
	AR("AR", "Arkansas"),
	AZ("AZ", "Arizona"),
	CA("CA", "California"),
	CO("CO", "Colorado"),
	CT("CT", "Connecticut"),
	//DC("DC", "the District of Columbia"),
	DE("DE", "Delaware"),
	FL("FL", "Florida"),
	GA("GA", "Georgia"),
	HI("HI", "Hawaii"),
	IA("IA", "Iowa"),
	ID("ID", "Idaho"),
	IL("IL", "Illinois"),
	IN("IN", "Indiana"),
	KS("KS", "Kansas"),
	KY("KY", "Kentucky"),
	LA("LA", "Louisiana"),
	MA("MA", "Massachusetts"),
	MD("MD", "Maryland"),
	ME("ME", "Maine"),
	MI("MI", "Michigan"),
	MN("MN", "Minnesota"),
	MO("MO", "Missouri"),
	MS("MS", "Mississippi"),
	MT("MT", "Montana"),
	NC("NC", "North Carolina"),
	ND("ND", "North Dakota"),
	NE("NE", "Nebraska"),
	NH("NH", "New Hampshire"),
	NJ("NJ", "New Jersey"),
	NM("NM", "New Mexico"),
	NV("NV", "Nevada"),
	NY("NY", "New York"),
	OH("OH", "Ohio"),
	OK("OK", "Oklahoma"),
	OR("OR", "Oregon"),
	PA("PA", "Pennsylvania"),
	RI("RI", "Rhode Island"),
	SC("SC", "South Carolina"),
	SD("SD", "South Dakota"),
	TN("TN", "Tennessee"),
	TX("TX", "Texas"),
	UT("UT", "Utah"),
	VA("VA", "Virginia"),
	VT("VT", "Vermont"),
	WA("WA", "Washington"),
	WI("WI", "Wisconsin"),
	WV("WV", "West Virginia"),
	WY("WY", "Wyoming");
	
	private String keyAbbreviation;
	private String valueFullname;
	
	US_STATES(String keyAbbreviation, String valueFullname) {
		this.keyAbbreviation = keyAbbreviation;
		this.valueFullname = valueFullname;
	}
	
	public String getKeyAbbreviation() {
		return keyAbbreviation;
	}
	
	public String getValueFullname() {
		return valueFullname;
	}
	
	public static US_STATES parse(String input) {
		if (input == null) {
			return null;
		}
	
		input = input.trim();	
		for (US_STATES state : values()) {
			if (state.keyAbbreviation.equalsIgnoreCase(input) || state.valueFullname.equalsIgnoreCase(input)) {
				return state;
			}
		}
	
		return null;
	}
}

File: ETL-Stream-Python/com/aaa/etl/us_states.py
'''
Created on 2023. 1. 21.

@author: joseph
'''
from enum import Enum

class US_STATES(Enum):
    AK="Alaska"
    AL="Alabama"
    AR="Arkansas"
    AZ="Arizona"
    CA="California"
    CO="Colorado"
    CT="Connecticut"
    #DC="the District of Columbia"
    DE="Delaware"
    FL="Florida"
    GA="Georgia"
    HI="Hawaii"
    IA="Iowa"
    ID="Idaho"
    IL="Illinois"
    IN="Indiana"
    KS="Kansas"
    KY="Kentucky"
    LA="Louisiana"
    MA="Massachusetts"
    MD="Maryland"
    ME="Maine"
    MI="Michigan"
    MN="Minnesota"
    MO="Missouri"
    MS="Mississippi"
    MT="Montana"
    NC="North Carolina"
    ND="North Dakota"
    NE="Nebraska"
    NH="New Hampshire"
    NJ="New Jersey"
    NM="New Mexico"
    NV="Nevada"
    NY="New York"
    OH="Ohio"
    OK="Oklahoma"
    OR="Oregon"
    PA="Pennsylvania"
    RI="Rhode Island"
    SC="South Carolina"
    SD="South Dakota"
    TN="Tennessee"
    TX="Texas"
    UT="Utah"
    VA="Virginia"
    VT="Vermont"
    WA="Washington"
    WI="Wisconsin"
    WV="West Virginia"
    WY="Wyoming"

    @staticmethod
    def parse(input_str):
        list_state = [i.value for i in US_STATES if i.name.lower() == input_str.lower() or i.value.lower() == input_str.lower()]
        if list_state.__len__() == 1:
            return list_state.pop(0)